pipeline/common/datasets.py

```diff
@@ -127,40 +127,35 @@ def shuffle_with_max_lines(
         if len(lines) == max_lines:
             break
 
-    # random.shuffle requires random access via indexing
-    # deque supports fast adding/removing from its ends with O(1)
-    # but indexing is O(N) which is too slow for shuffling large arrays
-    lines_list = list(lines)
-    lines = None
-    random.shuffle(lines_list)
+    line_index = len(lines)
```

Comment:
Was this retested on a large corpus to make sure we don't have an efficiency issue again?

Reply:
Just ran it on a 381 million sentences, and it seems fine:

```
[task 2024-08-30T02:19:32.575Z] [merge-mono] Wrote line 290,000,000 to /builds/worker/artifacts/mono.en.zst
[task 2024-08-30T02:21:47.752Z] [memory] 83.3 GB (+3.7 MB)
```